File: src/main/java/org/shark/example/domain/user/adapter/outbound/persistence/UserRepositoryImpl.java
package org.shark.example.domain.user.adapter.outbound.persistence;

import lombok.RequiredArgsConstructor;
import org.shark.example.domain.user.model.UserDto;
import org.shark.example.domain.user.model.UserRepository;
import org.springframework.stereotype.Repository;

@RequiredArgsConstructor
@Repository
public class UserRepositoryImpl implements UserRepository {

    private final UserJpaRepository userJpaRepository;

    @Override
    public void save(UserDto user) {
        UserDo userDo = new UserDo();
        userDo.setId(user.getId());
        userDo.setName(user.getName());
        userDo.setEmail(user.getEmail());
        userJpaRepository.save(userDo);
    }

    @Override
    public boolean existsByEmail(String email) {
        return userJpaRepository.existsByEmail(email);
    }
}

File: src/main/kotlin/org/shark/example/domain/user/adapter/outbound/persistence/UserRepositoryImpl.kt
package org.shark.example.domain.user.adapter.outbound.persistence

import org.shark.example.domain.user.model.UserDto
import org.shark.example.domain.user.model.UserRepository
import org.springframework.stereotype.Repository

@Repository
class UserRepositoryImpl(
    private val userJpaRepository: UserJpaRepository
) : UserRepository {

    override fun save(user: UserDto) {
        val userDo = UserDo()
        userDo.id = user.id
        userDo.name = user.name
        userDo.email = user.email
        userJpaRepository.save(userDo)
    }

    override fun existsByEmail(email: String): Boolean {
        return userJpaRepository.existsByEmail(email)
    }
}
